larray/core/array.py

```diff
@@ -6591,7 +6591,7 @@ def split_axis(self, axis, sep='_', names=None, regex=None):
          a0   0   1   2
          a1   3   4   5
         """
-        return self.reshape(self.axes.split_axis(axis, sep, names, regex))
+        return self.sort_axis(axis).reshape(self.axes.split_axis(axis, sep, names, regex))
 
```

Comment:
why not handle this as an option which is True by default. Users may unsorted axis after split?

Reply:
because it mangles the data and I do not see that as an interesting option ;-)

Reply:
OK